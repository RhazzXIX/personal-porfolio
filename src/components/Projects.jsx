import React from "react";
import "../styles/Projects.css";
import picSrc from "../assets/images/pBG.jpg";

export default function Projects() {
  const projects = [
    {
      imgSrc: picSrc,
      name: "Photo-tagging-app",
      description:
        'A "Where is Waldo" like App, where you need to look for a list of characters on the image. A project using firebase as a backend service.',
      id: 1,
    },
    {
      imgSrc: picSrc,
      name: "Shopping cart",
      description:
        "An E-commerce project. Used DummyJSON API as a simulation for backend communication. React-router-dom for a client side routing",
      id: 2,
    },
    {
      imgSrc: picSrc,
      name: "Memory Card",
      description:
        "A game app where to test your memory. Utilizing React Library functional components.",
      id: 3,
    },
    {
      imgSrc: picSrc,
      name: "Battleship",
      description:
        'A game app based on the board game "Battleship". Created using only Javascript, HTML & CSS',
      id: 4,
    },
    {
      imgSrc: picSrc,
      name: "Weather App",
      description:
        'An app that shows the current weather of the entered. Portrays my ability to communicate with the backend and handling "Promises"',
      id: 5,
    },
    {
      imgSrc: picSrc,
      name: "Restaurant",
      description:
        "A Restaurant page. A project showcasing webpack, ES6 Modules",
      id: 6,
    },
  ];

  return (
    <>
      <h2>Projects</h2>
      <ul className="cardContainer">
        {projects.map((project) => {
          return (
            <li className="card" key={project.id}>
              <img src={project.imgSrc} />
              <h3>{project.name}</h3>
              <p>{project.description}</p>
            </li>
          );
        })}
      </ul>
    </>
  );
}
