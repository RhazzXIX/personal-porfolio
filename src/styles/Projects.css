section#projects {
  margin-top: 60px;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

ul.cardContainer {
  margin-top: 20px;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  list-style: none;
  gap: 24px;
}

li.card {
  width: 300px;
  display: flex;
  flex-direction: column;
  background-color: #415a77;
  box-shadow: 0 0 4px black;
}

li.card > h3,
li.card > p {
  margin: 12px;
}
