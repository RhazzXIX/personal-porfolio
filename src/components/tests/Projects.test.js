import React from "react";
import { render } from "@testing-library/react";
import Projects from "../Projects";

describe("Projects component", () => {
  it("Renders on screen", () => {
    const { container } = render(<Projects />);

    expect(container).toMatchSnapshot();
  });
});
