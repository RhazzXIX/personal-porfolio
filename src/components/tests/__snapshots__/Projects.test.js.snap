// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component Renders on screen 1`] = `
<div>
  <h2>
    Projects
  </h2>
  <ul
    class="cardContainer"
  >
    <li
      class="card"
    >
      <img
        src="test-file-stub"
      />
      <h3>
        Photo-tagging-app
      </h3>
      <p>
        A "Where is Waldo" like App, where you need to look for a list of characters on the image. A project using firebase as a backend service.
      </p>
    </li>
    <li
      class="card"
    >
      <img
        src="test-file-stub"
      />
      <h3>
        Shopping cart
      </h3>
      <p>
        An E-commerce project. Used DummyJSON API as a simulation for backend communication. React-router-dom for a client side routing
      </p>
    </li>
    <li
      class="card"
    >
      <img
        src="test-file-stub"
      />
      <h3>
        Memory Card
      </h3>
      <p>
        A game app where to test your memory. Utilizing React Library functional components.
      </p>
    </li>
    <li
      class="card"
    >
      <img
        src="test-file-stub"
      />
      <h3>
        Battleship
      </h3>
      <p>
        A game app based on the board game "Battleship". Created using only Javascript, HTML & CSS
      </p>
    </li>
    <li
      class="card"
    >
      <img
        src="test-file-stub"
      />
      <h3>
        Weather App
      </h3>
      <p>
        An app that shows the current weather of the entered. Portrays my ability to communicate with the backend and handling "Promises"
      </p>
    </li>
    <li
      class="card"
    >
      <img
        src="test-file-stub"
      />
      <h3>
        Restaurant
      </h3>
      <p>
        A Restaurant page. A project showcasing webpack, ES6 Modules
      </p>
    </li>
  </ul>
</div>
`;
